### ISSUE TITLE
Controller:WaterCoil with Control Variable of HumidityRatio throws false node setpoint warning

### ISSUE THREAD

**Issue by mjwitte** [NREL/EnergyPlus#4958]
Helpdesk ticket 10513
User file with this chilled water coil controller

```
Controller:WaterCoil,
    DOAS DesCoil Controller, !- Name
    HumidityRatio,           !- Control Variable
    Reverse,                 !- Action
    Flow,                    !- Actuator Variable
    DesCoil AirOutlet,       !- Sensor Node Name
    DesCoil ChWInlet,        !- Actuator Node Name
```

Throws this warning, which appears to be false, because the simulation proceeds and appears to control close to the desired humidity ratio setpoint:

```
   ** Warning ** CheckForSensorAndSetpointNode: Coil:Cooling:Water="DESCOIL". 
   **   ~~~   **  ..Humidity ratio setpoint not found on coil air outlet node.
   **   ~~~   **  ..The setpoint may have been placed on a node downstream of the coil or on an airloop outlet node.
   **   ~~~   **  ..Specify the setpoint and the sensor on the coil air outlet node when possible.
   **   ~~~   **  ..Humidity ratio setpoint not found on coil air outlet node.
   **   ~~~   **  ..The setpoint may have been placed on a node downstream of the coil or on an airloop outlet node.
   **   ~~~   **  ..Specify the setpoint and the sensor on the coil air outlet node when possible.
```

**Comment by Nigusse** [comment]
@mjwitte @Myoldmopar This warning message was added to indicate to users that sensor node is at the airloop outlet node or downstream of the cooling coil instead of at the cooling coil outlet node. When the cooling coil sensor node is at the airloop outlet node, there was a case that the heating coils may turn on when it should have been off.  We found such cases, and changed the sensor nodes from the airloop outlet node to the cooling coil outlet node for all example files (more than 100, I think). This was done some time in 2013.  I will dig my emails and forward the email exchange or find the CR#.

**Comment by mjwitte** [comment]
@Nigusse But in the defect file, the setpoint is already at the coil outlet.  Unless I'm missing something, the test file should not be throwing the warning.

**Comment by Nigusse** [comment]
I will investigate further.

**Comment by Nigusse** [comment]
@mjwitte @Myoldmopar The warning is thrown because of mismatch of control variable type between the control variable specified in "Controller:WaterCoil" object and the setpoint manager object.  The control variable type specified in "SetpointManager:Scheduled" was "HumidityRatio".  Setpoint manager treats "HumidityRatio" and "MaximumHumidityRatio" as distinct types.  On the other hand "Controller:WaterCoil" allows "HumidityRatio" as an input but internally assumes "iHumidityRatioMaxSetPoint".

I am running modified defect files to confirm what happens when "MaximumHumidityRatio" is specified in "SetpointManager:Scheduled".  I expect no warning.  But this file runs forever even sizing period run only.

**Comment by Nigusse** [comment]
@mjwitte I added a check. If there is mismatch in the control variable type, then it resets the control variable type to "MaximumHumidityRatio" and issues a warning message but the simulation continuous.  The reset impacts the false warning message only. Sample warning message:

   *\* Warning *\* ResetHumidityRatioCtrlVarType: SetpointManager:Scheduled="DOAS DESCOIL HUMRATSPMANAGER". 
   *\*   ~~~   *\*  ..Humidity ratio control variable type specified is = HumidityRatio
   *\*   ~~~   *\*  ..Humidity ratio control variable type allowed with water coils is = MaximumHumidityRatio
   *\*   ~~~   *\*  ..Setpointmanager control variable type is reset to = MaximumHumidityRatio
   *\*   ~~~   *\*  ..Simulation continues.

**Comment by mjwitte** [comment]
Closed via #5146
